src/zinit/mod.rs

```diff
@@ -0,0 +1,311 @@
+const DEFAULT_ZINIT_PATH: &'static str = "/etc/zinit";
+
+#[derive(Deserialize)]
+struct CommandResult {
+    state: State,
+    body: Value,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum State {
+    Ok,
+    Error,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceStatus {
+    pub name: String,
+    pub pid: u64,
+    pub state: ServiceState,
+    pub target: ServiceTarget,
+    pub after: Vec<String>,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceState {
+    pub state: PossibleState,
+    pub reason: String,
+}
+
+#[derive(Deserialize, PartialEq)]
+pub enum PossibleState {
+    // ServiceStateUnknown is return when we cannot determine the status of a service
+    ServiceStateUnknown,
+    // ServiceStateRunning is return when we a service process is running and healthy
+    ServiceStateRunning,
+    // ServiceStateBlocked  is returned if the service can't start because of an unsatisfied dependency
+    ServiceStateBlocked,
+    // ServiceStateSpawned service has started, but zinit is not sure about its status yet.
+    // this is usually a short-lived status, unless a test command is provided. In that case
+    // the spawned state will only go to success if the test pass
+    ServiceStateSpawned,
+    // ServiceStateSuccess is return when a one shot service exited without errors
+    ServiceStateSuccess,
+    // ServiceStateError is return when we a service exit with an error (exit code != 0)
+    ServiceStateError,
+    //ServiceStateFailure is set of zinit can not spawn a service in the first place
+    //due to a missing executable for example. Unlike `error` which is returned if the
+    //service itself exits with an error.
+    ServiceStateFailure,
+}
+
+impl ServiceState {
+    fn exited(&self) -> bool {
+        matches!(
+            self.state,
+            PossibleState::ServiceStateBlocked
+                | PossibleState::ServiceStateSuccess
+                | PossibleState::ServiceStateFailure
+                | PossibleState::ServiceStateError
+        )
+    }
+
+    // fn is(&self, state: PossibleState) -> bool {
+    //     self.state == state
+    // }
+
+    fn any(&self, states: Vec<PossibleState>) -> bool {
+        return states.contains(&self.state);
+    }
+}
+
+#[derive(Deserialize)]
+pub enum ServiceTarget {
+    Up,
+    Down,
+}
+
+#[derive(Deserialize)]
+pub struct InitService {
+    pub exec: String,
+    pub oneshot: bool,
+    pub test: String,
+    pub after: Vec<String>,
+    pub env: HashMap<String, String>,
+    pub log: LogType,
+}
+
+#[derive(Deserialize)]
+pub enum LogType {
+    #[serde(rename = "stdout")]
+    StdoutLogType,
+    #[serde(rename = "ring")]
+    RingLogType,
+    #[serde(rename = "none")]
+    NoneLogType,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid command")]
+    InvalidCommand,
+
+    #[error("starting service")]
+    StartingService,
```

Comment:
`StartingService` doesn't sound like an error it also doesn't tell what exactly is going on. or when this error will be raised.